hypershift-operator/controllers/externalinfracluster/externalinfracluster_controller.go

```diff
@@ -112,20 +109,23 @@ func (r *ExternalInfraClusterReconciler) Reconcile(ctx context.Context, req ctrl
 	// TODO (alberto): populate the API and create/consume infrastructure via aws sdk
 	// role profile, sg, vpc, subnets.
 	if !hcp.Status.Ready {
-		r.Log.Info("Control plane is not ready yet. Requeuing")
-		return reconcile.Result{Requeue: true}, nil
+		r.Log.Info("hostedcontrolplane is not yet ready, requeueing")
+		return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
 	}
-
-	// Set the values for upper level controller
-	externalInfraCluster.Status.Ready = true
-	externalInfraCluster.Spec.ControlPlaneEndpoint = hyperv1.APIEndpoint{
-		Host: hcp.Status.ControlPlaneEndpoint.Host,
-		Port: hcp.Status.ControlPlaneEndpoint.Port,
-	}
-
-	if err := patchHelper.Patch(ctx, externalInfraCluster); err != nil {
-		r.Log.Error(err, "failed to patch")
-		return ctrl.Result{}, fmt.Errorf("failed to patch: %w", err)
+	if !externalInfraCluster.Status.Ready {
+		updated := externalInfraCluster.DeepCopy()
+		updated.Status.Ready = true
+		updated.Spec.ControlPlaneEndpoint = hyperv1.APIEndpoint{
+			Host: hcp.Status.ControlPlaneEndpoint.Host,
+			Port: hcp.Status.ControlPlaneEndpoint.Port,
+		}
+		err := r.Client.Status().Update(ctx, updated)
```

Comment:
can we `r.Client.Status().Patch`?

Reply:
We could but I don't see the point since there's nothing to merge — although looking more closely at this today I notice we're updating .spec and .status in the same call which makes me wonder if what I have here is missing something that isn't obvious. Going to do a couple of tests.